### PR TITLE
File Reorganization

### PR DESCRIPTION
**Pull request by kyle-stytch**
Linear Ticket: N/A

## Changes:

1. Reorganize the file structure of the SDK and combine some files. Kotlin recommended code style is too include similar classes together in the same file (https://kotlinlang.org/docs/coding-conventions.html#source-file-organization), rather than having one class per file as is required by Java.

1. Enable Kotlin Explict API mode (https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors), this forces everything publicly accessible to be marked `public` and given an explicit type. Good for ensuring I don't accidentally make something public that I didn't intend to.

1. Add .gitignore to ignore build files, some gradle files, some project configuration files

1. Add project code style settings file

1. Update various dependencies to more recent versions

1. Update text field in `StytchEditText` to expect an email address (this will disallow entering spaces in the text field)

1. Lots of minor formatting changes to conform to new code style settings

1. Add this PR template to the repo

## Notes:

- No changes to the public-facing SDK interface, except for a couple classes were moved to different packages (this would be a breaking change in the future, but for now with no one using the SDK it's not a problem)

- Next step for me after this is to write a bunch of unit tests

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
```
Comment (kyle-stytch) [inline gradle/wrapper/gradle-wrapper.properties:1]: Specify gradle version explicitly

File: .idea/codeStyles/Project.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (kyle-stytch) [inline .idea/codeStyles/Project.xml:1]: Code style settings file for Android Studio/IntelliJ

File: sdk/build.gradle [hunk 3]
```diff
@@ -67,6 +67,8 @@ android {
     }
     kotlinOptions {
         jvmTarget = '1.8'
+        // Enable Kotlin Explicit API mode: https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors
+        freeCompilerArgs += '-Xexplicit-api=strict'
```
Comment (kyle-stytch) [inline sdk/build.gradle:71]: Kotlin Explicit API mode

File: sdk/src/main/java/com/stytch/sdk/ui/view/StytchEditText.kt [hunk 4]
```diff
@@ -1,42 +1,47 @@
...
             background = getBackgroundShape(
                 config.uiCustomization.inputBackgroundColor.getColor(context),
                 config.uiCustomization.inputBackgroundBorderColor.getColor(context),
                 config.uiCustomization.inputCornerRadius,
-                1.dp.toInt()
+                1.dp.toInt(),
             )
         }
+        if (forEmailAddress) {
+            inputType = EditorInfo.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
```
Comment (kyle-stytch) [inline sdk/src/main/java/com/stytch/sdk/ui/view/StytchEditText.kt:43]: Set `EditText` input type to expect an email address

**Review by ethan-stytch — APPROVED** [review]
Looks good to me.

**Pull request merged**